CleverPush/Source/CPAppBannerModuleInstance.m

```diff
@@ -280,21 +280,26 @@ - (BOOL)bannerTargetingAllowed:(CPAppBanner*)banner {
     }
 
     NSString* appVersion = [[NSBundle mainBundle].infoDictionary valueForKey:@"CFBundleShortVersionString"];
-    allowed = [self checkRelationFilter:appVersion compareWith:banner.appVersionFilterValue relation:banner.appVersionFilterRelation isAllowed:allowed compareWithFrom:banner.fromVersion compareWithTo:banner.toVersion];
+    allowed = [self checkRelationFilter:appVersion compareWith:banner.appVersionFilterValue relation:banner.appVersionFilterRelation isAllowed:allowed compareWithFrom:banner.fromVersion compareWithTo:banner.toVersion languages:banner.languages];
     return allowed;
 }
 
 #pragma mark - check the banner triggering allowed as per selected version match with app version or not.
-- (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue relation:(NSString*)relation isAllowed:(BOOL)allowed compareWithFrom:(NSString*)compareValueFrom compareWithTo:(NSString*)compareValueTo {
-    return [self checkRelationFilter:value compareWith:compareValue compareWithFrom:compareValue compareWithTo:compareValue relation:relation isAllowed:allowed];
+- (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue relation:(NSString*)relation isAllowed:(BOOL)allowed compareWithFrom:(NSString*)compareValueFrom compareWithTo:(NSString*)compareValueTo languages:(NSMutableArray*)languagesValue {
+    return [self checkRelationFilter:value compareWith:compareValue compareWithFrom:compareValue compareWithTo:compareValue relation:relation isAllowed:allowed languages:languagesValue];
 }
 
 #pragma mark - check the banner triggering allowed as per selected version match with app version or not.
-- (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue compareWithFrom:(NSString*)compareValueFrom compareWithTo:(NSString*)compareValueTo relation:(NSString*)relation isAllowed:(BOOL)allowed {
+- (BOOL)checkRelationFilter:(NSString*)value compareWith:(NSString*)compareValue compareWithFrom:(NSString*)compareValueFrom compareWithTo:(NSString*)compareValueTo relation:(NSString*)relation isAllowed:(BOOL)allowed languages:(NSMutableArray*)languagesValue{
     
     if (relation == nil || compareValue == nil) {
         return allowed;
     }
+    if (languagesValue.count > 0 && [NSLocale preferredLanguages].count > 0) {
```

Comment:
Please do the `languages` check directly inside the `bannerTargetingAllowed` method as `checkRelationFilter` is only for appVersion filter.